ws/session.nim

```diff
@@ -184,7 +184,7 @@ proc handleClose*(
       raise newException(WSInvalidUTF8,
         "Invalid UTF8 sequence detected in close reason")
 
-  trace "Handling close message", code, reason
+  trace "Handling close message", code = ord(code), reason
```

Comment:
we probably want the string name here, no?

Reply:
Maybe not convinient for unknown codes?

Reply:
Above, `RangeError` is a `Defect`  - it cannot be caught reliably - enums must be **manually** checked for range using a case statement

Reply:
Range errors (conceptually) is precisely where making a defect catchable makes sense, in other words is the only excuse to make defect catchable. Now, it should have always been it's own class of errors and not defect, but that's a whole' nother story.

Reply:
Btw I totally agree, defect should be catchable, it just sucks that we can't handle range errors in an uncluttered way.

Reply:
>  defect catchable

There's no hard or consistent rule for what should be a defect and what shouldn't, it depends on the context ("in my API, this is a Defect") - conceptually, there's no first principle to lean back on except your mood in the morning when you write the code, so it'll always be prone to bikeshedding - such is the language however, and this along with range checking of integer conversions, null pointer checks and overflows are all things you have to check manually, every time - probably though, it would be trivial to write a macro that verifies the values of an enum and returns an Option or Result - I suspect using such a macro would be even less cluttered than the exception solution.

Reply:
@jangko @dryajov @Menduist this btw is something btw that the code needs auditing for across the codebase - I often have to repeat it in reviews which errors are Defects so I suspect there's more of them - `RangeError` for example is an old name, it's called `RangeDefect` in newer version: https://github.com/nim-lang/Nim/blob/v1.4.8/lib/system/exceptions.nim#L156